### PR TITLE
[RFC] storage: custom storage volume snapshot API

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

This is intended for API review but requires further work.

### CODE REVIEW

**Comment by brauner** [comment]
@monstermunchkin, this is the API. It should have everything it needs. However, internally a few things need to be adjusted:
- when copying and moving custom storage volumes snaphots need to be copied too
- renaming storage volumes with snapshots needs to be handled

File: shared/api/storage_pool_volume_snapshot.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	Name string `json:"name" yaml:"name"`
+}
+
+// StorageVolumeSnapshot represents a LXD storage volume snapshot
+//
+// API extension: storage_api_volume_snapshots
+type StorageVolumeSnapshot struct {
+	Name        string            `json:"name" yaml:"name"`
+	Config      map[string]string `json:"config" yaml:"config"`
+	Description string            `json:"description" yaml:"description"`
```
Comment (stgraber) [inline shared/api/storage_pool_volume_snapshot.go]: We're missing a way to update this field it looks like.

For this we'd need:
 - UpdateStorageVolumeSnapshot
 - PUT /1.0/storage-pools/POOL/TYPE/VOLUME/snapshots/NAME handler
 - StorageVolumeSnapshotPut struct (with just Description in it)

Comment (brauner) [reply shared/api/storage_pool_volume_snapshot.go]: Added. :)

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by brauner** [comment]
Closing so that @monstermunchkin can work on it.

**Pull request closed without merging**